### PR TITLE
Fix fail2ban010

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5943

we need to add a new script to destroy the ipset when shorewall stops, but `shorewall restart` does not exactly the same things than `systemctl restart shorewall`, IPSET complains that the set are still used by the kernel, so I need to use systemctl with `nethserver-fail2ban-save` when the sysadmin add or remove the jails.

Now we create the bantime with ipset on the fly, but we cannot have anymore a dynamic bantime for recidive because we are limited to `2147483` which is the maximum timeout of ipset. If you prefer we could set the recidive to the maximum, it is about 24 days

### CODE REVIEW

File: root/etc/e-smith/templates/etc/fail2ban/jail.local/10recidive [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
     use NethServer::Fail2Ban;
-    my $bantime  = $fail2ban{BanTime}*336 || '604800';
-    my $perpetual  = $fail2ban{Recidive_Perpetual} || 'disabled';
-    $bantime  = '0' if ($perpetual eq 'enabled');
     my $findtime = '86400';
+    my $bantime  = '1209600';
+    my $perpetual  = $fail2ban{Recidive_Perpetual} || 'disabled';
+    $bantime  = '-1' if ($perpetual eq 'enabled');
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/fail2ban/jail.local/10recidive]: We already changed from `-1` to `0`. Why we go again for `-1`?

File: root/etc/e-smith/templates/etc/shorewall/initdone/20fail2ban [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+    return "" if ($fail2ban{'status'} ne 'enabled');
+    my $bantime= $fail2ban{'BanTime'} || '1800';
+    my $perpetual = $fail2ban{'Recidive_Perpetual'} || 'disabled';
     foreach (NethServer::Fail2Ban::listAllJails()) {
-        $OUT .= "system(\"/usr/sbin/ipset -quiet -exist create f2b-$_ hash:ip timeout 600 \");\n";
+    
+        if (($_ eq 'recidive') && ($perpetual eq 'enabled')) {
+            $OUT .= "system(\"/usr/sbin/ipset -quiet -exist create f2b-recidive hash:ip timeout 0 \");\n";
+        } elsif ($_ eq 'recidive') {
+            # max ban time for ipset is 2147483 seconds
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/shorewall/initdone/20fail2ban]: Why the comment says `2147483` but when creating the ipset the timeout is set to `1209600`?

File: root/etc/e-smith/templates/etc/shorewall/stopped/20base [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/shorewall/stopped/20base]: Please rename this fragment to `20fail2ban`

File: root/usr/libexec/nethserver/fail2ban-ipset-destroy [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+# destroy all f2b-X when shorewall stop
+
+for j in $(ipset -L -name | grep '^f2b' | cut -d'-' -f2-)
```
Comment (gsanchietti) [inline root/usr/libexec/nethserver/fail2ban-ipset-destroy]: You could avoid the call to `cut`:
```bash
for j in $(ipset -L -name | grep '^f2b')
do
    /usr/sbin/ipset  -quiet destroy $j
done
```

File: root/etc/e-smith/templates/etc/shorewall/initdone/20fail2ban (hunk 2 above)
Comment (stephdl) [reply root/etc/e-smith/templates/etc/shorewall/initdone/20fail2ban]: the max timeout is 2147483 in ipset if you try to be higher you have an error
so either we set to 24 days (2147483) or the max ban we allow in the UI (1209600)


got to the max of the UI

File: root/usr/libexec/nethserver/fail2ban-ipset-destroy (hunk 4 above)
Comment (stephdl) [reply root/usr/libexec/nethserver/fail2ban-ipset-destroy]: done

File: root/etc/e-smith/templates/etc/shorewall/stopped/20base (hunk 3 above)
Comment (stephdl) [reply root/etc/e-smith/templates/etc/shorewall/stopped/20base]: done

**Pull request merged**